datastream-server/src/main/java/com/linkedin/datastream/server/CoordinatorEventBlockingQueue.java

```diff
@@ -46,8 +46,11 @@ public synchronized void put(CoordinatorEvent event) {
       }
     }
 
-    // always overwrite the event in the map
-    _eventMap.put(event.getType(), event);
+    // we put into the eventMap for dedup only if there is no metadata
+    if (event.getEventMetadata() == null) {
```

Comment:
We can go ahead with this for time being. In the long term we probably want the dedup logic to take metadata into consideration. With this implementation we have inconsistency that only events without metadata get deduped. Also there is no way for someone to know about this behavior unless checking out the code.

Reply:
Deduping eventMetadata will requires assumptions to have the metadata properly implement the hashcode. This should be covered in the future implementation.